### PR TITLE
Clarify `std::env::vars(_os)` documentation

### PR DESCRIPTION
**Pull request by poliorcetics**
Are the tests ok for public documentation or should they be moved into `src/env/tests.rs` ?

### CODE REVIEW

File: library/std/src/env.rs [hunk 1]
```diff
@@ -92,7 +92,9 @@ pub struct VarsOs {
 ///
 /// The returned iterator contains a snapshot of the process's environment
 /// variables at the time of this invocation. Modifications to environment
-/// variables afterwards will not be reflected in the returned iterator.
+/// variable values afterwards will not be reflected in the returned iterator
+/// but modifications to the environment using [`set_var`] and [`remove_var`]
+/// will appear.
```
Comment (the8472) [inline library/std/src/env.rs:97]: This change suggests that using `set_var/remove_var` will change which items an already created iterator will return.

File: library/std/src/env.rs [hunk 2]
```diff
@@ -112,6 +114,19 @@ pub struct VarsOs {
...
 /// ```
 ///
+/// ```
+/// use std::env;
+///
+/// let before_set = env::vars().count();
+/// env::set_var("TEST_ADD_REMOVE_ENV_VAR", "42");
+/// let before_remove = env::vars().count();
+/// env::remove_var("TEST_ADD_REMOVE_ENV_VAR");
+/// let after_remove = env::vars().count();
```
Comment (the8472) [inline library/std/src/env.rs:124]: But this only creates iterators and immediately calculates their counts, it does not exercise the snapshotting in any form.

**Review by the8472 — COMMENTED** [review]
The doctest does not seem to reflect the wording.

**Comment by poliorcetics** [comment]
I completely misunderstood the docs, sorry ! I thought the whole env was snapshotted at the start of the program, not at the iterator creation, my bad.

**Pull request closed without merging**